### PR TITLE
Kfdef converter

### PR DESCRIPTION
**Pull request by gabrielwen**
Relates to #4002 
Adds a new function to read from a file path and returns KfDef in v1beta1 format.  If the file path points to v1alpha1 KfDef, will converts to v1beta1.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4149)

### CODE REVIEW

**Comment by gabrielwen** [comment]
@jlewi @yanniszark @swiftdiaries @kkasravi this PR is ready.

File: bootstrap/pkg/apis/apps/group.go [hunk 1]
```diff
@@ -155,6 +155,14 @@ const (
 	EXISTING_ARRIKTO = "existing_arrikto"
 )
 
+// Plugin kind used starting from v1beta1
+const (
+	AWS_PLUGIN_KIND              = "KfAwsPlugin"
+	GCP_PLUGIN_KIND              = "KfGcpPlugin"
+	MINIKUBE_PLUGIN_KIND         = "KfMinikubePlugin"
+	EXISTING_ARRIKTO_PLUGIN_KIND = "KfExistingArriktoPlugin"
+)
```
Comment (swiftdiaries) [inline bootstrap/pkg/apis/apps/group.go]: Hey Gabriel
Thanks for the this!
What about KfDefs for the kfctl_k8s? How would they be converted?

File: bootstrap/pkg/apis/apps/kfdef/kfloader.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+	if err := yaml.Unmarshal(configs, kfdef); err != nil {
+		return nil, &kfapis.KfError{
+			Code:    int(kfapis.INVALID_ARGUMENT),
+			Message: fmt.Sprintf("invalid config file format: %v", err),
+		}
+	}
+	return kfdef, nil
+}
+
+// Loads KfDef fild and returns KfDef in v1beta1. v1alpha1 will be converted to the new version.
```
Comment (swiftdiaries) [inline bootstrap/pkg/apis/apps/kfdef/kfloader.go]: nit: fild -> file

File: bootstrap/pkg/apis/apps/kfdef/kfloader.go [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+func LoadKfDefFromURI(configFile string) (*kfdefv1beta1.KfDef, error) {
+	if configFile == "" {
```
Comment (swiftdiaries) [inline bootstrap/pkg/apis/apps/kfdef/kfloader.go]: So any KfDef config would be converted to the v1beta1 and then used?

Comment (gabrielwen) [reply bootstrap/pkg/apis/apps/kfdef/kfloader.go]: yes, after this call all KfDef should be in v1beta1.

File: bootstrap/pkg/apis/apps/group.go (hunk 1 above)
Comment (gabrielwen) [reply bootstrap/pkg/apis/apps/group.go]: this is only used for plugins.  from this file it doesn't have any plugins?
https://github.com/kubeflow/manifests/blob/master/kfdef/kfctl_k8s_istio.yaml

File: bootstrap/pkg/apis/apps/kfdef/kfloader.go (hunk 2 above)
Comment (gabrielwen) [reply bootstrap/pkg/apis/apps/kfdef/kfloader.go]: fixed.

**Comment by jlewi** [comment]
/lgtm
/approve
/hold

Added a hold in case anyone else needs to LGTM this; on my end its ready to go.

**Comment by gabrielwen** [comment]
/hold cancel

sorry I need this for reconcile semantics.  let me know if anyone thinks this PR needs to be updated.  I can make follow up changes.

**Pull request merged**